### PR TITLE
Add Dockerfile for Stackdriver Logging Agent container image.

### PR DESCRIPTION
**Pull request by qingling128**

### CODE REVIEW

**Comment by qingling128** [comment]
This is a follow up of https://github.com/GoogleCloudPlatform/google-fluentd/pull/42.

PTAL

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
+FROM ubuntu:xenial
+
+RUN apt-get update \
+    && apt-get install -y \
+        curl \
+        lsb-release \
+    && rm -rf /var/lib/apt/lists/* \
```
Comment (dhrupadb) [inline docker/Dockerfile]: This should be done last (after all apt-get commands). I believe auto remove depends on these files to some extent.

File: docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    && curl -sS https://dl.google.com/cloudagents/install-logging-agent.sh | bash \
+    && /opt/google-fluentd/embedded/bin/gem uninstall fluent-plugin-google-cloud \
+    && /opt/google-fluentd/embedded/bin/gem install fluent-plugin-google-cloud -v 0.6.8.pre.1 \
+    && apt-get autoremove \
+    && sed -i "s/num_threads 8/num_threads 8\n  detect_json true\n  # Enable metadata agent lookups.\n  enable_metadata_agent true\n  metadata_agent_url \"http:\/\/$METADATA_AGENT_HOSTNAME:8000\"/" "/etc/google-fluentd/google-fluentd.conf"
+
+ENV METADATA_AGENT_HOSTNAME ""
```
Comment (bmoyles0117) [inline docker/Dockerfile]: We can remove this environment variable I believe, and set the default value in the config to the internal DNS.

File: docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+RUN apt-get update \
+    && apt-get install -y \
+        curl \
+        lsb-release \
+    && rm -rf /var/lib/apt/lists/* \
+    && curl -sS https://dl.google.com/cloudagents/install-logging-agent.sh | bash \
+    && /opt/google-fluentd/embedded/bin/gem uninstall fluent-plugin-google-cloud \
+    && /opt/google-fluentd/embedded/bin/gem install fluent-plugin-google-cloud -v 0.6.8.pre.1 \
+    && apt-get autoremove \
+    && sed -i "s/num_threads 8/num_threads 8\n  detect_json true\n  # Enable metadata agent lookups.\n  enable_metadata_agent true\n  metadata_agent_url \"http:\/\/$METADATA_AGENT_HOSTNAME:8000\"/" "/etc/google-fluentd/google-fluentd.conf"
```
Comment (bmoyles0117) [inline docker/Dockerfile]: $METADATA_AGENT_HOSTNAME gets replaced onbuild here based on a --build-arg, not at runtime based on $ENV

File: docker/Dockerfile (hunk 1 above)
Comment (bmoyles0117) [reply docker/Dockerfile]: Sounds reasonable.

File: docker/Dockerfile (hunk 3 above)
Comment (qingling128) [reply docker/Dockerfile]: Something like `$ENV['METADATA_AGENT_HOSTNAME']`?

Comment (qingling128) [reply docker/Dockerfile]: What about non-GKE environment (like Docker)?

File: docker/Dockerfile (hunk 1 above)
Comment (qingling128) [reply docker/Dockerfile]: Changed.

File: docker/Dockerfile (hunk 3 above)
Comment (qingling128) [reply docker/Dockerfile]: Per discussion offline, we will do using the static `local-metadata-agent.stackdriver.com:8000` in the image. GKE, Docker or other systems will sed replace the URL as seen fit.

**Comment by qingling128** [comment]
Feedback addressed. PTAL.

File: docker/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,14 @@
+FROM ubuntu:xenial
+
+RUN apt-get update \
+    && apt-get install -y \
+        curl \
+        lsb-release \
+    && curl -sS https://dl.google.com/cloudagents/install-logging-agent.sh | bash \
+    && /opt/google-fluentd/embedded/bin/gem uninstall fluent-plugin-google-cloud \
+    && /opt/google-fluentd/embedded/bin/gem install fluent-plugin-google-cloud -v 0.6.8 \
```
Comment (igorpeshansky) [inline docker/Dockerfile]: Do we want to just run `gem install` here and let it pick up the latest released gem version? Or is there a good reason to freeze the version?

Comment (qingling128) [reply docker/Dockerfile]: In fact, I'm thinking about bumping the `fluent-plugin-google-cloud` version to `0.6.8` in `google-fluentd` and release a new package some time next week. After that we don't need the `gem uninstall` and `gem install` stuff.

WDYT?

Comment (igorpeshansky) [reply docker/Dockerfile]: I think the decision we need to make is whether we're tracking the latest released package, or the latest released gem. The latter would let us push the container image ahead of the package, but I'm not sure how much that buys us. It seems that you're leaning towards the former.

Comment (qingling128) [reply docker/Dockerfile]: Yeah, I'm leaning towards the former. If the user really wants, they can always build their custom image and bump the gem version there.

WDYT?

Comment (igorpeshansky) [reply docker/Dockerfile]: SGTM. So let's remove this from here and release 1.5.19 instead?

Comment (qingling128) [reply docker/Dockerfile]: Removed.

**Comment by qingling128** [comment]
PTAL.

**Pull request merged**